src/controllers/edgeconnect/deployment/deployment.go

```diff
@@ -88,11 +89,18 @@ func prepareContainerEnvVars(instance *edgeconnectv1alpha1.EdgeConnect) []corev1
 }
 
 func buildAppLabels(instance *edgeconnectv1alpha1.EdgeConnect) *kubeobjects.AppLabels {
+	version := "latest"
+	statusVersion := instance.Status.Version.Version
+
+	if statusVersion != "" && len(validation.IsQualifiedName(statusVersion)) == 0 {
```

Comment:
We should just take whatever is in the status, and not validate it again. Similar with the empty check, we should default the version status to `latest` and use it as tag verbatim, without introducing duplicate logic here.

Reply:
But it's not implemented, I created a ticket for that, we don't have code which reads LABEL.